packages/core/src/InlineEditor/InlineEditor.tsx

```diff
@@ -20,7 +20,7 @@ export { staticClasses as inlineEditorClasses };
 
 export type HvInlineEditorClasses = ExtractNames<typeof useClasses>;
 
-export interface HvInlineEditorProps
+export interface HvInlineEditorProps<T extends React.ElementType>
```

Comment:
Really nice 🙏 I just think we should probably implement this like the other polymorphic components we have: button, typography,  time ago, etc.